### ISSUE TITLE
"Open the repository in your external editor" not working for Atom and Xcode (different problems)

### ISSUE THREAD

**Issue by dhhepting** [desktop/desktop#9155]
### Describe the bug

When I select "Open the repository in your external editor" with my external editor set to Atom, the repository is not opened and the following error dialogue box appears (username blacked out in picture):

![Screen Shot 2020-02-21 at 1 05 48 PM](https://user-images.githubusercontent.com/11279142/75064502-b4a45c80-54ac-11ea-83c2-904f234c130a.png)

This issue may be related to https://github.com/desktop/desktop/issues/9050, regarding BBEdit. I am not able to change the path for Atom.
Atom does open properly when invoked from Finder.

On my system, I have the option in Github Desktop preferences to select Xcode as my external editor. When I choose Xcode as my external editor and try to open the repository with it,
nothing happens.

I've looked for error messages in Console, but haven't found any - perhaps I am not looking in the right places.

### Version & OS

GitHub Desktop 2.3.1
macOS 10.14.6
Atom 1.44.0
Xcode 11.0 (11A420a)

### Steps to reproduce the behavior

Starting screen:
![Screen Shot 2020-02-21 at 1 49 05 PM](https://user-images.githubusercontent.com/11279142/75067053-c9cfba00-54b1-11ea-90c1-e2530949ec57.png)

1. Go to 'Open the repository in your external editor'
2. Click on 'Open in Atom'
3. See error (dialogue box with error shown above and below)

### Expected behavior

I expect the Atom editor to open when I click "Open in Atom" and Xcode to open when I click "Open in Xcode" (depending on my external editor setting in preferences)

### Actual behavior

When I click "Open in Atom", I see this error dialogue:

![Screen Shot 2020-02-21 at 1 05 48 PM](https://user-images.githubusercontent.com/11279142/75064502-b4a45c80-54ac-11ea-83c2-904f234c130a.png)

and this line is generated in 2020-02-21.desktop.production.log:
`2020-02-21T20:08:13.277Z - info: [ui] Current tutorial step is NotApplicable`

When I click "Open in Xcode", nothing happens but this line is generated in 2020-02-21.desktop.production.log:
`2020-02-21T20:08:32.728Z - info: [ui] Current tutorial step is NotApplicable`

**Comment by tierninho** [comment]
@dhhepting thanks for reporting this.

Try moving Atom and Xcode to the Applications folder and then restart Desktop. Let us know how that goes.

**Comment by dhhepting** [comment]
@tierninho thanks for the quick reply!

Atom:
- was in `/Applications`
- I removed `~/Library/Caches/com.github.atom.ShipIt/`
- I restarted GitHub Desktop
- Success!

Xcode:
- was in `/Applications`
- there is a file called `com.apple.dt.Xcode` in the `~/Library/Caches/`directory. I tried to open it but got an error dialogue that said the file cannot be opened because it is missing its project.pbxproj file.
- I removed the file
- No change in behaviour 

I'm happy to have resolved the issue with Atom. It seems that my update of Atom did not go or was not done according to plan, but removing the cache (`~/Library/Caches/com.github.atom.ShipIt/`) seemed to do the trick.

**Comment by tierninho** [comment]
Glad the Atom issue is resolved!! The `com.apple.dt.Xcode` file for Xcode appears safe to delete, however not sure what is happening with Desktop. For reference, here is the recent pull request as we just added Xcoe as an editor option: https://github.com/desktop/desktop/pull/8255

@msztech Any chance you recognize what's going on here? Much appreciated.

**Comment by msztech** [comment]
Hi @dhhepting @tierninho  Could you try to open a file with `xed` in the command line. The pull request #8255 is essentially using the xed to open Xcode.

**Comment by dhhepting** [comment]
@msztech @tierninho Thanks for the suggestions. I couldn't run `xed` at first as I got this error message: `xcode-select: error: tool 'xed' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance`. 

I found a solution for this at https://stackoverflow.com/questions/17980759/xcode-select-active-developer-directory-error and now `xed` runs from the command line and the "Open in Xcode" button also works -- except when the repository is a directory, when Xcode reports that it cannot open the file (which is a directory, not a file).

I've tried `xed repo_dir/*` which seems to do better, but it tries to open every file in the repository and reports when it can't open a file that is a directory. Trying `xed repo_dir/README.md` works to open Xcode without error but doesn't get all the repository's contents and internal directory structure. As far as I can see, the argument to `xed` needs to be one or more files.
